File: Rentify.Services/Database/Reservation.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Rentify.Services.Database
{
    public class Reservation
    {
        [Key]
        public int Id { get; set; }

        [ForeignKey(nameof(UserId))]
        public int UserId { get; set; }
        public User? User { get; set; }

        [ForeignKey(nameof(PropertyId))]
        public int PropertyId { get; set; }
        public Property? Property { get; set; }

        public bool IsMonthly { get; set; }

        [ForeignKey(nameof(StatusId))]
        public int StatusId { get; set; } = 1;
        public Status? Status { get; set; }

        public DateTime CreatedAt { get; set; }
        public DateTime? StartDateOfRenting { get; set; }
        public DateTime? EndDateOfRenting { get; set; }
    }
}

File: UI/rentify_desktop/lib/models/reservation.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:rentify_desktop/models/property.dart';
import 'package:rentify_desktop/models/status.dart';
import 'package:rentify_desktop/models/user.dart';

part 'reservation.g.dart';

@JsonSerializable()
class Reservation {
  final int id;
  final int userId;
  final User? user;
  final int propertyId;
  final Property? property;
  final bool isMonthly;
  final int statusId;
  Status? status;
  DateTime? createdAt;
  DateTime? startDateOfRenting;
  DateTime? endDateOfRenting;

  Reservation({
    required this.id,
    required this.userId,
    required this.user,
    required this.propertyId,
    required this.property,
    required this.isMonthly,
    required this.statusId,
    this.status,
    this.createdAt,
    this.startDateOfRenting,
    this.endDateOfRenting,
  });

  factory Reservation.fromJson(Map<String, dynamic> json) =>
      _$ReservationFromJson(json);

  Map<String, dynamic> toJson() => _$ReservationToJson(this);
}
